server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -893,12 +903,15 @@ private static double objectToDouble(Object value) {
 
         private final NumberType type;
 
-        public NumberFieldType(NumberType type) {
-            super();
+        public NumberFieldType(String name, NumberType type, boolean isSearchable, boolean hasDocValues, Map<String, String> meta) {
+            super(name, isSearchable, hasDocValues, meta);
             this.type = Objects.requireNonNull(type);
-            setTokenized(false);
-            setHasDocValues(true);
-            setOmitNorms(true);
+            this.setIndexAnalyzer(Lucene.KEYWORD_ANALYZER);     // allows number fields in significant text aggs - do we need this?
```

Comment:
I don't think we meant to support it, no. We probably should deprecate it and drop it in 8.0.

I think it is probably worth adding a comment that this won't cause us to make an index of the text representation of the keyword.